tiledb/api/src/array/fragment_info.rs

```diff
@@ -0,0 +1,667 @@
+use std::ops::Deref;
+
+use anyhow::anyhow;
+// use serde::{Deserialize, Serialize};
+// use serde_json::json;
+
+use crate::array::schema::{RawSchema, Schema};
+use crate::config::{Config, RawConfig};
+use crate::context::{CApiInterface, Context, ContextBound};
+use crate::datatype::Datatype;
+use crate::error::Error;
+use crate::fn_typed;
+use crate::key::LookupKey;
+use crate::non_empty_domain::{NonEmptyDomain, Range};
+use crate::string::{RawTDBString, TDBString};
+use crate::Result as TileDBResult;
+
+pub(crate) enum RawFragmentInfo {
+    Owned(*mut ffi::tiledb_fragment_info_t),
+}
+
+impl Deref for RawFragmentInfo {
+    type Target = *mut ffi::tiledb_fragment_info_t;
+    fn deref(&self) -> &Self::Target {
+        let RawFragmentInfo::Owned(ref ffi) = *self;
+        ffi
+    }
+}
+
+impl Drop for RawFragmentInfo {
+    fn drop(&mut self) {
+        let RawFragmentInfo::Owned(ref mut ffi) = *self;
+        unsafe {
+            ffi::tiledb_fragment_info_free(ffi);
+        }
+    }
+}
+
+pub struct FragmentInfo<'ctx> {
+    context: &'ctx Context,
+    raw: RawFragmentInfo,
+}
+
+impl<'ctx> ContextBound<'ctx> for FragmentInfo<'ctx> {
+    fn context(&self) -> &'ctx Context {
+        self.context
+    }
+}
+
+impl<'ctx> FragmentInfo<'ctx> {
+    pub(crate) fn capi(&self) -> *mut ffi::tiledb_fragment_info_t {
+        *self.raw
+    }
+
+    pub fn config(&self) -> TileDBResult<Config> {
+        let mut c_config: *mut ffi::tiledb_config_t = out_ptr!();
+        self.capi_return(unsafe {
+            ffi::tiledb_fragment_info_get_config(
+                self.context.capi(),
+                self.capi(),
+                &mut c_config,
+            )
+        })?;
+
+        Ok(Config::from_raw(RawConfig::Owned(c_config)))
+    }
+
+    pub fn load(&self) -> TileDBResult<()> {
+        self.capi_return(unsafe {
+            ffi::tiledb_fragment_info_load(self.context.capi(), self.capi())
```

Comment:
Is this a required call for the fragment info to be useful?
If so it should be called in the build or new methods

Reply:
I waffled on this a bit because this can be a relatively heavy operation depending on the array. I'll go peek at what Python does for a second opinion.

Reply:
I ended up having `build()` automatically call `load()` and then added an extra `build_without_loading` just in case we end up needing it for some reason. I could see in async land that we might want to create the FragmentInfoList in one thread and then shunt it off to a thread pool to do the I/O or something similar.